ttm/manager.py

```diff
@@ -0,0 +1,659 @@
+            self.logger.debug('totest already publishing')
+            can_publish = False
+
+        if self.update_pinned_descr:
+            self.status_for_openqa = {
+                'current_snapshot': current_snapshot,
+                'new_snapshot': new_snapshot,
+                'snapshotable': snapshotable,
+                'can_release': can_release,
+                'is_publishing': totest_is_publishing,
+            }
+            self.update_openqa_status_message()
+
+        if can_publish:
+            if current_qa_version == current_snapshot:
+                self.publish_factory_totest()
+                self.write_version_to_dashboard('snapshot', current_snapshot)
+                can_release = False  # we have to wait
+            else:
+                # We reached a very bad status: openQA testing is 'done', but not of the same version
+                # currently in test project. This can happen when 'releasing' the
+                # product failed
+                raise Exception('Publishing stopped: tested version (%s) does not match version in test project (%s)'
+                                % (current_qa_version, current_snapshot))
+
+        if can_release:
+            self.update_totest(new_snapshot)
+            self.write_version_to_dashboard('totest', new_snapshot)
+
+
+    def release(self, project):
+        self.setup(project)
+        new_snapshot = self.current_sources()
+        self.update_totest(new_snapshot)
+
+    def write_version_to_dashboard(self, target, version):
+        version_file = 'version_%s' % target
+        if self.project.is_image_product:
+            version_file = version_file + '_images'
+        if not (self.dryrun or self.norelease):
+            self.api.pseudometa_file_ensure(version_file, version, comment='Update version')
+
+    def load_issues_to_ignore(self):
+        text = self.api.attribute_value_load('IgnoredIssues')
+        if text:
+            root = yaml.safe_load(text)
+            self.issues_to_ignore = root.get('last_seen')
+        else:
+            self.issues_to_ignore = dict()
+
+    def save_issues_to_ignore(self):
+        if self.dryrun:
+            return
+        text = yaml.dump({'last_seen': self.issues_to_ignore}, default_flow_style=False)
+        self.api.attribute_value_save('IgnoredIssues', text)
+
+    def release_version(self):
+        url = self.api.makeurl(['build', self.project.name, 'standard', self.project.arch,
+                                '000release-packages:%s-release' % self.project.base])
+        f = self.api.retried_GET(url)
+        root = ET.parse(f).getroot()
+        for binary in root.findall('binary'):
+            binary = binary.get('filename', '')
+            result = re.match(r'.*-([^-]*)-[^-]*.src.rpm', binary)
+            if result:
+                return result.group(1)
+
+        raise NotFoundException("can't find %s version" % self.project)
+
+    def current_sources(self):
```

Comment:
I don't like the "source" naming here - it should at least contain the word "version" in one way or another.

Reply:
I found there were too many versions - so I wanted something different than 'current_version'. Meanwhile I'd go with version_from_project